lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinition.cs

```diff
@@ -27,23 +27,139 @@ namespace Org.Apache.REEF.IMRU.API
     /// <seealso cref="IMRUJobDefinitionBuilder" />
     public sealed class IMRUJobDefinition
     {
-        private readonly IConfiguration _configuration;
         private readonly string _jobName;
+        private readonly IConfiguration _mapFunctionConfiguration;
+        private readonly IConfiguration _mapInputCodecConfiguration;
+        private readonly IConfiguration _updateFunctionCodecsConfiguration;
+        private readonly IConfiguration _reduceFunctionConfiguration;
+        private readonly IConfiguration _updateFunctionConfiguration;
+        private readonly IConfiguration _mapOutputPipelineDataConverterConfiguration;
+        private readonly IConfiguration _mapInputPipelineDataConverterConfiguration;
+        private readonly IConfiguration _partitionedDatasetConfiguration;
+        private readonly int _numberOfMappers;
 
-        internal IMRUJobDefinition(IConfiguration configuration, string jobName)
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="mapFunctionConfiguration">Map function configuration</param>
+        /// <param name="mapInputCodecConfiguration">Map input codec configuration</param>
+        /// <param name="updateFunctionCodecsConfiguration">codec configuration for update 
+        /// function. It is union of TMapInput, TMapOutput and TResult configuration</param>
+        /// <param name="reduceFunctionConfiguration">Reduce function configuration</param>
+        /// <param name="updateFunctionConfiguration">Update function configuration</param>
+        /// <param name="mapOutputPipelineDataConverterConfiguration">Configuration of 
+        /// PipelineDataConverter for TMapOutput</param>
+        /// <param name="mapInputPipelineDataConverterConfiguration">Configuration of 
+        /// PipelineDataConverter for TMapInput</param>
+        /// <param name="partitionedDatasetConfiguration">Configuration of partitioned 
+        /// dataset</param>
+        /// <param name="numberOfMappers">Number of mappers</param>
+        /// <param name="jobName">Job name</param>
+        internal IMRUJobDefinition(
```

Comment:
And i thought about it. it's going to happen frequently
On Aug 5, 2015 10:03, "Markus Weimer" <notifications@github.com> wrote:

> In lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinition.cs
> <https://github.com/apache/incubator-reef/pull/334#discussion_r36326142>:
>
> > +        /// </summary>
> > +        /// <param name="mapFunctionConfiguration">Map function configuration</param>
> > +        /// <param name="mapInputCodecConfiguration">Map input codec configuration</param>
> > +        /// <param name="updateFunctionCodecsConfiguration">codec configuration for update
> > +        /// function. It is union of TMapInput, TMapOutput and TResult configuration</param>
> > +        /// <param name="reduceFunctionConfiguration">Reduce function configuration</param>
> > +        /// <param name="updateFunctionConfiguration">Update function configuration</param>
> > +        /// <param name="mapOutputPipelineDataConverterConfiguration">Configuration of
> > +        /// PipelineDataConverter for TMapOutput</param>
> > +        /// <param name="mapInputPipelineDataConverterConfiguration">Configuration of
> > +        /// PipelineDataConverter for TMapInput</param>
> > +        /// <param name="partitionedDatasetConfiguration">Configuration of partitioned
> > +        /// dataset</param>
> > +        /// <param name="numberOfMappers">Number of mappers</param>
> > +        /// <param name="jobName">Job name</param>
> > +        internal IMRUJobDefinition(
>
> Ah, I think I finally got your concern. The question is not whether we can
> bind the same codec implementation for TMapInput and TMapOutput. We can
> do that by using named parameters. Your concern is: What if this codec has
> additional named parameters, which have to be bound to different values?
> That indeed is impossible, in the absence of REEF-31
> <https://issues.apache.org/jira/browse/REEF-31>.
>
> Did I understand correctly?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-reef/pull/334/files#r36326142>.
>